main.tf

```diff
@@ -15,19 +15,31 @@ locals {
   iam_role_name = var.use_existing_iam_role ? var.iam_role_name : (
     length(var.iam_role_name) > 0 ? var.iam_role_name : "${var.prefix}-iam-${random_id.uniq.hex}"
   )
-  mfa_delete = var.bucket_enable_versioning && var.bucket_enable_mfa_delete
+  mfa_delete                = var.bucket_enable_versioning && var.bucket_enable_mfa_delete
+  bucket_encryption_enabled = var.bucket_enable_encryption && var.bucket_encryption_enabled
+  bucket_logs_enabled       = var.bucket_logs_enabled && var.bucket_enable_logs
+  bucket_versioning_enabled = var.bucket_enable_versioning && var.bucket_versioning_enabled
 }
 
 resource "random_id" "uniq" {
   byte_length = 4
 }
 
+resource "aws_kms_key" "lacework_kms_key" {
+  count                   = (var.use_existing_cloudtrail || length(var.bucket_sse_key_arn) > 0) ? 0 : 1
+  description             = "A KMS key used to encrypt CloudTrail logs which are monitored by Lacework"
+  deletion_window_in_days = var.kms_key_deletion_days
+  multi_region            = var.kms_key_multi_region
+  tags                    = var.tags
+  policy                  = data.aws_iam_policy_document.kms_key_policy.json
+}
+
 resource "aws_cloudtrail" "lacework_cloudtrail" {
   count                      = var.use_existing_cloudtrail ? 0 : 1
   name                       = var.cloudtrail_name
   is_multi_region_trail      = true
   s3_bucket_name             = local.bucket_name
-  kms_key_id                 = var.bucket_sse_key_arn
+  kms_key_id                 = aws_kms_key.lacework_kms_key[0].arn
```

Comment:
Did some further testing and this will cause an error in the case of a user creating a new cloudtrail, but bringing their own key.

Perhaps we want to add a local `bucket_sse_key_arn` and have it be something like:
```hcl
bucket_sse_key arn = (var.use_existing_cloudtrail || length(var.bucket_sse_key_arn) > 0) ? var.bucket_sse_key_arn : aws_kms_key.lacework_kms_key[0].arn
```